src/libraries/System.Runtime.InteropServices.JavaScript/gen/JSImportGenerator/JSImportStubContext.cs

```diff
@@ -118,12 +127,11 @@ static bool IsSkipLocalsInitAttribute(AttributeData a)
             };
             int typesHash = Math.Abs((int)hash);
 
-
-
             var fullName = $"{method.ContainingType.ToDisplayString()}.{method.Name}";
             string qualifiedName;
+
             var ns = string.Join(".", method.ContainingType.ToDisplayParts().Where(p => p.Kind == SymbolDisplayPartKind.NamespaceName).Select(x => x.ToString()).ToArray());
-            var cn = string.Join("/", method.ContainingType.ToDisplayParts().Where(p => p.Kind == SymbolDisplayPartKind.ClassName).Select(x => x.ToString()).ToArray());
+            var cn = string.Join("/", method.ContainingType.ToDisplayParts().Where(p => Array.IndexOf(nameKinds, p.Kind) >= 0).Select(x => x.ToString()).ToArray());
             var qclasses = method.ContainingType.ContainingNamespace == null ? ns : ns + "." + cn;
             qualifiedName = $"[{env.Compilation.AssemblyName}]{qclasses}:{method.Name}";
```

Comment:
This would be better without linq, and multiple string constructions. IIUC, the first few parts would be namespace, and then will have all class/struct/etc kinds. If so, then we can use a single StringBuilder, loop through to add the namespace parts, then append the rest with `/`'s.